### PR TITLE
feat: raised phpstan level and fix errors

### PR DESCRIPTION
**Pull request by ta-tikoma**
Detailed typehints

### CODE REVIEW

**Comment by nunomaduro** [comment]
@ta-tikoma Let me review this tonight, so I can ensure it does not break anything on the Pest's arch plugin.

File: src/Elements/ObjectDescription.php [hunk 1]
```diff
@@ -6,9 +6,6 @@
 
 use PHPUnit\Architecture\Asserts\Properties\ObjectPropertiesDescription;
 
-/**
- * Open to extends
- */
-class ObjectDescription extends ObjectPropertiesDescription
+final class ObjectDescription extends ObjectPropertiesDescription
```
Comment (nunomaduro) [inline src/Elements/ObjectDescription.php:9]: Can you not make this class final? Or maybe simply add the `@final` annotation please?

**Review by nunomaduro — CHANGES_REQUESTED** [review]
Ok! Seems to be working good with the Pest plugin. Except that comment about the new `final` class.

File: src/Elements/ObjectDescription.php (hunk 1 above)
Comment (ta-tikoma) [reply src/Elements/ObjectDescription.php]: complete

**Pull request merged**